posts/2020-03-15-docs-rs-opt-into-fewer-targets.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: "docs.rs requests your help in reducing build costs"
+author: Joshua Nelson
+team: the docs.rs team <https://www.rust-lang.org/governance/teams/dev-tools#docs-rs>
+---
+
+Recently, [docs.rs] added a feature that allows crates to opt-out of building on all targets.
+If you don't need to build on all targets, we ask that you enable this feature
+to allow us to reduce queue times and storage costs.
+
+## What does the feature do?
+
+By default, docs.rs builds all crates published to [crates.io] for [every tier one target][metadata].
+However, most crates have the same content on all target.
+Of the platform-dependent crates, almost all target a single platform,
+and do not need to be built on other targets.
+For example, [`winapi`] only has documentation on the `x86_64-pc-windows-msvc`
+and `i686-pc-windows-msvc` targets, and is blank on all others.
+
+This feature allows you to request building only on specific targets.
+For example, [`winapi`] could opt into only building windows targets
+by putting the following in its `Cargo.toml`:
+
+```toml
+[package.metadata.docs.rs]
+# This also sets the default target to `x86_64-pc-windows-msvc`
+targets = ["x86_64-pc-windows-msvc", "i686-pc-windows-msvc"]
+```
+
+If you only need a single target, it's even simpler:
+
+```toml
+[package.metadata.docs.rs]
+# This builds only the default target (usually `x86_64-unknown-linux-gnu`)
+targets = []
```

Comment:
I would not recommend using the empty target list. While it's supported to prevent unnecessary build failures, we should encourage people to fill the default target in if they use `targets`.

Reply:
I think this will help increase adoption by making it easier to opt-in, though. If we change it to suggest `targets = ["x86_64-unknown-linux-gnu"]`, then someone might type it by hand and spell it wrong, failing the build for no reason.

Reply:
I don't think this will prevent adoption too much, as people will copy/paste this snippet anyway. Also, your misspelling concern would apply to every other target people could add.

Reply:
Ok, I'll change it to `targets = ["x86_64-unknown-linux-gnu"]`.